src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/AnsiStringMarshaller.cs

```diff
@@ -72,15 +76,19 @@ public AnsiStringMarshaller(string? str, Span<byte> buffer)
         /// <remarks>
         /// <seealso cref="CustomTypeMarshallerFeatures.TwoStageMarshalling"/>
         /// </remarks>
-        public void FromNativeValue(byte* value) => _allocated = value;
+        public void FromNativeValue(byte* value)
+        {
+            _nativeValue = value;
```

Comment:
Shouldn't this call `FreeNative` before overwriting `_nativevalue`? Otherwise it could leak any memory allocated in the constructor when the parameter is passed byref.

Reply:
The source generator expects this method to abandon the ownership of the current `_nativeValue`, without freeing it. I agree that it is confusing. We may want to have tweak the names or have more methods for different cases to make the expected semantics clear. @dotnet/interop-contrib Thoughts?

Reply:
The names were debated ad nauseum in API review, so I'd be more inclined to add more if that is needed. I'm open to other method conventions if that would clear things up though.

Reply:
The names were debated ad nauseum in API review, so I'd be more inclined to add more if that is needed. I'm open to other method conventions if that would clear things up though.

Reply:
I think adding more won't clear the confusion about the semantics of `FromNativeValue`. `FromNativeValue` is meant to overwrite the whole marshaller state that totally not obvious from the name. https://github.com/dotnet/runtime/issues/69096 that was just opened is rooted in the same confusion.

Let's continue this discussion in #69096 and #69031.